optimism/src/folding.rs

```diff
@@ -276,17 +289,30 @@ mod tests {
             }));
 
         // Check conversion of general expressions
-        let xyz = x * y * z;
+        let xyz = x.clone() * y * z;
         let xyz_f = FoldingCompatibleExpr::<TestConfig>::BinOp(
             Op2::Mul,
             Box::new(FoldingCompatibleExpr::<TestConfig>::BinOp(
                 Op2::Mul,
-                Box::new(x_f),
+                Box::new(x_f.clone()),
                 Box::new(y_f),
             )),
             Box::new(z_f),
         );
-
         assert_eq!(FoldingCompatibleExpr::<TestConfig>::from(xyz), xyz_f);
+
+        let x_endo = x + endo;
+        let x_endo_f = FoldingCompatibleExpr::<TestConfig>::BinOp(
```

Comment:
So we either add it directly as a constant, or we use the hardcoded `EndoCoefficient`, and checking if these are the same. And the `as_literal` is just for convenience of conversion?

Reply:
Yes